### ISSUE TITLE
X/Y plot broken

### ISSUE THREAD

**Issue by tildebyte** [LucianoCirino/efficiency-nodes-comfyui#42]
Image generation works as expected and completes, but just before final image output, I get the traceback below
```
XY Plot Results:
img_count: 30
img_dims: 768 x 512
plot_dim: 10 x 3
ckpt: 
clip_skip: -1
lora(mod,clip): 000001.safetensors(1.0,1.0), 000002.safetensors(1.0,1.0), 000003.safetensors(1.0,1.0), 000004.safetensors(1.0,1.0),  000005.safetensors(1.0,1.0),  000006.safetensors(1.0,1.0),  000007.safetensors(1.0,1.0),  000008.safetensors(1.0,1.0),  000009.safetensors(1.0,1.0)
vae: 
seed: 599774203296300
steps: 30
cfg: 6.0, 7.0, 8.0
sampler: dpmpp_2m
scheduler: karras
denoise: 1.0
!!! Exception during processing !!!
```
```python
Traceback (most recent call last):
  File "D:\ComfyUI\execution.py", line 145, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\ComfyUI\execution.py", line 75, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\ComfyUI\execution.py", line 68, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\ComfyUI\custom_nodes\Efficiency\efficiency_nodes.py", line 1433, in sample
    font_size = adjusted_font_size(text, initial_font_size, img.width)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\ComfyUI\custom_nodes\Efficiency\efficiency_nodes.py", line 1314, in adjusted_font_size
    text_width, _ = font.getsize(text)
                    ^^^^^^^^^^^^
AttributeError: 'FreeTypeFont' object has no attribute 'getsize'
```

I tried working around the above, but get *new* TBs and decided I didn't want to end up as a maintainer in here.

Rolling back to [this](https://github.com/LEv145/images-grid-comfy-plugin/blob/main/workflows/efficiency.png) X/Y Plot node (as soon as I find its commit), given that it actually works

**Comment by tildebyte** [comment]
...except that the old X/Y node had no support for LoRAs, so I guess I'll just have to wait

**Comment by LucianoCirino** [comment]
Are you using the most updated version of the Efficiency Nodes from Github with the most up to date version of ComfyUI? There are some changes he made to Comfy that forced me to make some changes to how LoRAs are loaded from my nodes.

**Comment by tildebyte** [comment]
Yes, but I don't see what that would have to do with `AttributeError: 'FreeTypeFont' object has no attribute 'getsize'` while trying to render labels for a finished grid of images.

**Comment by LucianoCirino** [comment]
can you please show me your workflow?

**Comment by tildebyte** [comment]
Only removed the prompt and the name of the actual LoRA [example_json.txt](https://github.com/LucianoCirino/efficiency-nodes-comfyui/files/11996481/example_json.txt).

Working around the first error netted me errors in `text_width, text_height = d.textsize(text, font=font)` something about instance of 'd' not having a `textsize` method, which is dumb, given that it's a 'PIL.ImageDraw', and I have PIL (==10.x) installed in the venv I'm using... I got past both occurrences of that with the \<sarcasm>amazingly brilliant workaround of `text_width, text_height = 10, 10`\<sarcasm/> (yes, I are a Softweird Engineer ;p)

**Comment by jteijema** [comment]
```console
got prompt
----------------------------------------
XY Plot Script Inputs:
(X) CFG Scale: [7]
(Y) Nothing: ['']
----------------------------------------
100%|███████████████████████████████████████████████████████████████████████████████████| 20/20 [00:01<00:00, 17.51it/s]
----------------------------------------
XY Plot Results:
img_count: 1
img_dims: 512 x 512
plot_dim: 1 x 1
ckpt: radiant_vibes.safetensors
clip_skip: -1
lora(mod,clip): Sci-fi_Enviroments.safetensors(1.0000000000000009,1.0)
vae: Baked VAE
seed: 0
steps: 20
cfg: 7
sampler: euler_ancestral
scheduler: karras
denoise: 1.0
!!! Exception during processing !!!
Traceback (most recent call last):
  File "/home/ubuntu/ComfyUI/execution.py", line 145, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
  File "/home/ubuntu/ComfyUI/execution.py", line 75, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
  File "/home/ubuntu/ComfyUI/execution.py", line 68, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
  File "/home/ubuntu/ComfyUI/custom_nodes/efficiency-nodes-comfyui/efficiency_nodes.py", line 1433, in sample
    font_size = adjusted_font_size(text, initial_font_size, img.width)
  File "/home/ubuntu/ComfyUI/custom_nodes/efficiency-nodes-comfyui/efficiency_nodes.py", line 1314, in adjusted_font_size
    text_width, _ = font.getsize(text)
AttributeError: 'FreeTypeFont' object has no attribute 'getsize'

Prompt executed in 2.91 seconds
```

Same issue for me

**Comment by jteijema** [comment]
[See 10.0.0 ](https://github.com/LucianoCirino/efficiency-nodes-comfyui/)

<html><body>


Removed | Use instead
-- | --
FreeTypeFont.getsize() and FreeTypeFont.getoffset() | FreeTypeFont.getbbox() and FreeTypeFont.getlength()


</body>
</html>

**Comment by jteijema** [comment]
GetLength will soft issue for FreeTypeFont but ImageDraw.textsize() is also removed.

**Comment by LucianoCirino** [comment]
Thanks @jteijema. Pull request accepted. Try again @tildebyte.

**Comment by tildebyte** [comment]
> but ImageDraw.textsize() is also removed.

For posterity, (IIUC) @jteijema replaced it with a call to `ImageDraw.textbbox()`

**Comment by LucianoCirino** [comment]
> > but ImageDraw.textsize() is also removed.
> 
> For posterity, (IIUC) @jteijema replaced it with a call to `ImageDraw.textbbox()`

Is issue resolved? can this item be closed?

**Comment by tildebyte** [comment]
Sorry, I haven't had a chance to verify it yet, so it's completely up to you

**Comment by tildebyte** [comment]
Fixed with #43
